database/consensus.go

```diff
@@ -336,7 +337,20 @@ WHERE date = $4`
 	return nil
 }
 
+func (db *Db) SetTxSender(tx *junotypes.Tx) error {
+	stmt := `UPDATE transaction SET sender = $1 WHERE hash LIKE $2`
+
+	hash := tx.TxHash[:len(tx.TxHash)-1] + "%"
+
+	_, err := db.Sqlx.Exec(stmt, tx.GetSigners()[0].String(), hash)
```

Comment:
you are accessing to first element before checking it exists

Reply:
But each transaction has at least one signer